Client/Application/AppDelegate.swift

```diff
@@ -74,8 +74,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Ecosia: lifecycle tracking
         Analytics.shared.activity(.launch)
         
-        // Ecosia: Engagement Service Initialization helper
-        ClientEngagementService.shared.initializeAndUpdateNotificationRegistrationIfNeeded(notificationCenterDelegate: self)
+        /* 
+         Ecosia: Feature Management fetch
+         We perform the same configuration retrieval in
+         `applicationDidBecomeActive(:)` and sounds redundant;
+         However we need it here to make sure we retrieve the latest
+         flag state of the EngagementService.
+         Decouple the "loading" only from the filesystem of any
+         previously saved Model from the `Unleash.start(:)` will not
+         make any tangible difference in the process as we check if
+         any cached version of the Model is in place.
+         */
+        Task {
+            await FeatureManagement.fetchConfiguration()
+            // Ecosia: Engagement Service Initialization helper
+            ClientEngagementService.shared.initializeAndUpdateNotificationRegistrationIfNeeded(notificationCenterDelegate: self)
```

Comment:
Looks good to me! I'll guess that way we are on the safe side that we don't initialise Braze before Unleash flags are fetched.

Reply:
That's the intention yeah.
For now, it's the safest approach we can take, given that Unleash needs to be "fetched" every time we resume the app as well. Annoyingly this will mean that at launch time the `fetchConfgiration()` will be called twice but the cached model will be retrieved wihtout performing any further step.